plugins/modules/ec2_key.py

```diff
@@ -319,16 +356,26 @@ def main():
     name = module.params['name']
     state = module.params.get('state')
     key_material = module.params.get('key_material')
-    force = module.params.get('force')
     key_type = module.params.get('key_type')
+    tags = module.params.get('tags')
+
+    result = {}
 
     if key_type:
         module.require_botocore_at_least('1.21.23', reason='to set the key_type for a keypair')
 
     if state == 'absent':
-        delete_key_pair(module, ec2_client, name)
+        result = delete_key_pair(module, ec2_client, name)
+
     elif state == 'present':
-        create_key_pair(module, ec2_client, name, key_material, force, key_type)
+        # check if key already exists
+        key = find_key_pair(ec2_client, name)
+        if key:
+            result = handle_existing_key_pair_update(module, ec2_client, name, key)
```

Comment:
I think we can avoid passing module in if we pass "module.params" and "module.check_mode" which should make testing easier

Reply:
not sure if we should not pass `module` to `handle_existing_key_pair_update` as it's being further passed to `ensure_ec2_tags` here https://github.com/ansible-collections/amazon.aws/pull/1288/files#diff-236420337d2eb886687a30f1923a5cf04cd290cb7064a43e0a93cdecefb78701R327-R329
If there's a way to do it that I might be missing, open to suggestions :)

Reply:
You're probably right, I think ensure_ec2_tags expects to be able to use module.fail_json_aws.